+++ ngx_http_upload_module.c	2015-04-26 12:55:27.548905458 +0000
@@ -1678,7 +1678,6 @@
     ngx_http_upload_merger_state_t ms;
     off_t        remaining;
     ssize_t      rc;
-    int          result;
     ngx_buf_t    in_buf;
     ngx_buf_t    out_buf;
     ngx_http_upload_loc_conf_t  *ulcf = ngx_http_get_module_loc_conf(u->request, ngx_http_upload_module);
@@ -1771,10 +1770,6 @@
         out_buf.file_pos += out_buf.last - out_buf.pos;
     }

-    if(out_buf.file_pos < state_file->info.st_size) {
-        result = ftruncate(state_file->fd, out_buf.file_pos);
-    }
-
     rc = ms.complete_ranges ? NGX_OK : NGX_AGAIN;

 failed: